Data/PackageData.cs

```diff
@@ -404,6 +404,12 @@ private static async Task<PackageData> ReadPackageFromUrl (PackageData package,
             {
                 //System.Console.WriteLine($"DOWNLOADING {package.DownloadUrl}");
                 var r = await httpClient.GetAsync (package.DownloadUrl, token).ConfigureAwait (false);
+                try {
+                    r.EnsureSuccessStatusCode ();
+                }
+                catch (Exception ex) {
+                    throw new Exception($"Failed to download {package.DownloadUrl}", ex);
+                }
```

Comment:
Instead of catch/throw you could do:

```suggestion
                if (!r.IsSuccessStatusCode) {
                    throw new Exception($"Failed to download {package.DownloadUrl} due to HTTP response status code {r.StatusCode}.");
                }
```